### PR TITLE
[fix](Nereids) non-inner join should not merge dist info

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

left join should use left dist info.
right join should use right dist info.
full outer join should return ANY dist info.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildOutputPropertyDeriver.java [hunk 1]
```diff
@@ -239,18 +239,28 @@ public PhysicalProperties visitPhysicalHashJoin(
...
+                case RIGHT_OUTER_JOIN:
+                    return new PhysicalProperties(rightHashSpec);
+                case FULL_OUTER_JOIN:
+                    return PhysicalProperties.ANY;
+                default:
+                    throw new AnalysisException("unknown join type " + hashJoin.getJoinType());
             }
-
             // shuffle, if left child is natural mean current join is bucket shuffle join
             // and remain natural for colocate join on upper join.
```
Comment (XieJiann) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/properties/ChildOutputPropertyDeriver.java:262]: This comment should be removed.

**Review by keanji-x — APPROVED** [review]

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.6 seconds
 stream load tsv:          538 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162102082 Bytes

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.86 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          99.5 seconds inserted 10000000 Rows, about 100K ops/s
 storage size: 17162281190 Bytes

**Review by starocean999 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**